File: leetcode/1easy/이경호/1614.cpp
class Solution {
public:
  int maxDepth(string s) {
    int max = 0;
    int d = 0;
    for (char c : s) {
      if (c == '(') {
        d++;
        max = d > max ? d : max;
      }
      if (c == ')') {
        d--;
      }
    }
    return max;
  }
};

File: leetcode/1easy/박은비/1614.swift
<<<<<<< HEAD
class Leetcode.이재훈.이재훈.Solution {
=======
class Leetcode.이재훈.Solution {
>>>>>>> ab7f46f73d8f18bcd38c5cf06ecabae64f53a424
    func maxDepth(_ s: String) -> Int {
/*
(1+(2*3)+((8)/4))+1

( -> depth를 증가
) -> depth를 감소


*/
        var maxCount: Int = 0
        var depth: Int = 0

        for c in s {
            switch c {
                case "(":
                    depth += 1
                    maxCount = max(maxCount, depth)
                case ")":
                    depth -= 1
                default:
                    continue
            }
        }


        return maxCount
    }
}
